From: Heiko Becker <heirecka@exherbo.org>
Date: Fri, 12 Feb 2016 12:37:15 +0100
Subject: [PATCH] Search for libsystemd first, fall back to libsystemd-journal

systemd >= 209 merged the individual libraries libsystemd-journal,
libsystemd-login, libsystemd-id128 and libsystemd-daemon into
a single library, libsystemd. To ease the transition one could pass
an option to its build to generate stub libraries and matching
pkg-config files. With systemd >= 229 this option has now been
removed, causing the build to fail when the journald option is
enabled.

Change-Id: I26670f207f1a9e79c16be5ce8c8a49353143c5ba
Reviewed-by: Oswald Buddenhagen <oswald.buddenhagen@theqtcompany.com>
Reviewed-by: Robin Burchell <robin.burchell@viroteck.net>
---
 qtbase/config.tests/unix/journald/journald.pro | 6 +++++-
 qtbase/src/corelib/global/global.pri           | 5 ++++-
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/qtbase/config.tests/unix/journald/journald.pro b/qtbase/config.tests/unix/journald/journald.pro
index 2bb50ceb71c5..ea765642e6ea 100644
--- a/qtbase/config.tests/unix/journald/journald.pro
+++ b/qtbase/config.tests/unix/journald/journald.pro
@@ -1,6 +1,10 @@
 SOURCES = journald.c
 
 CONFIG += link_pkgconfig
-PKGCONFIG_PRIVATE += libsystemd-journal
+
+packagesExist(libsystemd): \
+    PKGCONFIG_PRIVATE += libsystemd
+else: \
+    PKGCONFIG_PRIVATE += libsystemd-journal
 
 CONFIG -= qt
diff --git a/qtbase/src/corelib/global/global.pri b/qtbase/src/corelib/global/global.pri
index aa4945f90e3a..dd846955f61e 100644
--- a/qtbase/src/corelib/global/global.pri
+++ b/qtbase/src/corelib/global/global.pri
@@ -53,7 +53,10 @@ slog2 {
 
 journald {
     CONFIG += link_pkgconfig
-    PKGCONFIG_PRIVATE += libsystemd-journal
+    packagesExist(libsystemd): \
+        PKGCONFIG_PRIVATE += libsystemd
+    else: \
+        PKGCONFIG_PRIVATE += libsystemd-journal
     DEFINES += QT_USE_JOURNALD
 }
 
